plugins/mac-capture/mac-sck-audio-capture.m

```diff
@@ -8,14 +8,17 @@
 API_AVAILABLE(macos(13.0)) static void destroy_audio_screen_stream(struct screen_capture *sc)
 {
     if (sc->disp && !sc->capture_failed) {
+        os_event_t *disp_finished;
```

Comment:
I wonder, wouldn't it be sufficient in this case to:

* Retain local copies of the `sc->stream_properties` and `sc->disp` pointers
* Move the `release` call for both _into_ the completion handler
* Set both pointers inside the struct immediately to `NULL` after calling `stopCaptureWithCompletionHandler`

That way the entire synchronisation with an `os_event` becomes unnecessary.

Reply:
Looking at it more closely, I don't think either is necessary. We don't need any references to `sc->stream_properties`, `sc->tex` etc in the completion handler , and I saw that `stopCaptureWithCompletionHandler` increases the retainCount which suggests that the call holds its own reference until the completion handler is done (which makes sense).
As such, I believe we can simply call `stopCaptureWithCompletionHandler` without any synchronization or manual reference retaining, and simply move on. I will update the PR to remove the event.